.github/workflows/lint.yml

```diff
@@ -0,0 +1,23 @@
+name: Lint
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.x
+
+      - name: Install dependencies
+        run: pip install pre-commit
+
+      - name: Run pre-commit
+        run: pre-commit run --all-files --show-diff-on-failure
```

Comment:
@jooola FYI there's an official action for this. Actually, a pair of actions: https://pre-commit.ci/lite#example-using-pre-commit-action.

Besides, it's possible to enable the hosted service for this repo and then, it shouldn't be necessary to have a duplicate GHA workflow.

Reply:
Thanks for the links, I didn't know pre-commit.ci was working on a new GitHub action. I remember the old one being deprecated in favor of the pre-commit.ci service.

For our use case I think this is just fine, and once the new GitHub action is out of beta we might reconsider this.

Reply:
Why not just use pre-commit.ci? Using GHA seems like a waste in this case, especially since the GHA resources are shared across the entire org.

Reply:
I have no experience with pre-commit.ci, I implemented something that has been working for me for ages. But I am happy to start using it to reduce the GHA resources usage.

With that said, I recently stumbled on https://results.pre-commit.ci/run/github/244995547/1688396810.pSqkyRKhQqGD40pGmTdY1Q and I haven't figured out how to solve this issue yet.

I will take some time once I am done with the current PRs I am working on.